Lib/test/test_typing.py

```diff
@@ -4228,6 +4246,64 @@ def test_get_type_hints(self):
             {'a': typing.Optional[int], 'b': int}
         )
 
+    def test_get_type_hints_generic(self):
+        self.assertEqual(
+            get_type_hints(BarGeneric[int].__origin__), 
+            {'a': typing.Optional[T], 'b': int}
+        )
+
+    def test_generic_inheritance(self):
+        class A(TypedDict, Generic[T]):
+            a: T
+
+        self.assertEqual(A.__bases__, (Generic, dict,))
+        self.assertEqual(A.__parameters__, (T,))
+        self.assertEqual(A[str].__parameters__, ())
+        self.assertEqual(A[str].__args__, (str,))
+
+        class B(A[KT], total=False):
+            b: KT
+
+        self.assertEqual(B.__bases__, (Generic, dict,))
+        self.assertEqual(B.__total__, False)
+        self.assertEqual(B.__parameters__, (KT, ))
+        self.assertEqual(B.__optional_keys__, frozenset(['b']))
+        self.assertEqual(B.__required_keys__, frozenset(['a']))
+
+        self.assertEqual(B[str].__parameters__, ())
+        self.assertEqual(B[str].__args__, (str,))
+        self.assertEqual(B[str].__origin__, B)
+
+        class C(B[int]):
+            c: int
+
+        self.assertEqual(C.__bases__, (dict,))
+        self.assertEqual(C.__total__, True)
+        self.assertEqual(C.__optional_keys__, frozenset(['b']))
+        self.assertEqual(C.__required_keys__, frozenset(['a', 'c']))
+        assert C.__annotations__ == {
+            'a': T,
+            'b': KT,
+            'c': int,
+        }
+        # with self.assertRaises(TypeError):
+        #    C[str]
+
+        class WithImplicitAny(B):
```

Comment:
One more important pattern, that we have missed: adding new generic arguments in child classes.

```python
class Point2D(TypedDict, Generic[X, Y]):
    x: X
    y: Y 


class Point3D(Point2D[X, Y], Generic[X, Y, Z]):
    z: Z
```

This works for regular types, I guess that `TypedDict` should also work the same way.